### PR TITLE
Add repoVisibility for other-than-GitHub repos

### PR DESCRIPTION
**Pull request by n2ygk**
## Hey, I just made a Pull Request!

Fixes: #4313

- [x] add repoVisibility for GitLab repos analogous to that implemented for GitHub repos
- [x] ditto for BitBucket
- [ ] ditto for Azure

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

**Comment by n2ygk** [comment]
- Bitbucket [API doc](https://developer.atlassian.com/bitbucket/api/2/reference/resource/workspaces/%7Bworkspace%7D/projects#post) `"is_private": true|false`
- Azure Devops [API doc](https://docs.microsoft.com/en-us/rest/api/azure/devops/core/projects/create?view=azure-devops-rest-6.0#projectvisibility) `"ProjectVisibility ": "private"|"public"`??

**Comment by n2ygk** [comment]
FYI -- this **changes** the default visibility of GitLab and BitBucket repos to public, making them consistent with GitHub. This could be considered a breaking change. It's easy enough to switch the defaults to match the prior behavior if that's what's wanted.

**Comment by n2ygk** [comment]
Azure DevOps has separate Project visibility above Repo visibility so there's apparently no way to set a repo private or public as it inherits the visibility of the project: https://docs.microsoft.com/en-us/azure/devops/organizations/public/make-project-public

This is too different for me to figure out. I'll skip this for Azure and leave it to @benjdlambert, @jhaals, or @mfrinnstrom  with deeper Azure knowledge (since they touched azure.ts last ;-).

**Comment by n2ygk** [comment]
rebased to 8eb862c

File: .changeset/sour-gorillas-fail.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+To configure the repo visibility, set scaffolder._type_.visibility as in this example:
+
+```yaml
+scaffolder:
+  github:
+    visibility: private # 'public' or 'internal' or 'private' (default is 'public')
+  gitlab:
+    visibility: public # 'public' or 'internal' or 'private' (default is 'public')
+  bitbucket:
+    visibility: public # 'public' or 'private' (default is 'public')
```
Comment (freben) [inline .changeset/sour-gorillas-fail.md:22]: ooof i see the schema for the scaffolder leaves some things to be desired, but anyway, these should probably be added there https://github.com/backstage/backstage/blob/master/plugins/scaffolder-backend/config.d.ts#L18

**Review by freben — COMMENTED** [review]
Would you mind adding this to the schema as well? Besides that we're good to go I think.

File: .changeset/sour-gorillas-fail.md (hunk 1 above)
Comment (n2ygk) [reply .changeset/sour-gorillas-fail.md:22]: I don't quite understand what this entry for github is about:
```
      [key: string]: string;
```
But will give it a try

Comment (n2ygk) [reply .changeset/sour-gorillas-fail.md:22]: Yeah scaffolder schema does not support the integrations concepts of a list of hosts for each flavor, so things like `visibility` become global for all instances of a given remote repo flavor.

Comment (n2ygk) [reply .changeset/sour-gorillas-fail.md:22]: Got that wrong. Standby.

Comment (n2ygk) [reply .changeset/sour-gorillas-fail.md:22]: Let's try that again. Also rebased to 5df0714

Comment (freben) [reply .changeset/sour-gorillas-fail.md:22]: Yeah it's just a "match anything here" basically; at the time it was written somebody took a quick solution without defining the full field details :)

**Comment by n2ygk** [comment]
Sure.

File: plugins/scaffolder-backend/config.d.ts [hunk 2]
```diff
@@ -19,16 +19,19 @@ export interface Config {
   scaffolder?: {
     github?: {
       [key: string]: string;
+      visiblity?: string;
```
[the remark this answers is not in the document]
Comment (n2ygk) [reply plugins/scaffolder-backend/config.d.ts]: Weirdly I've done this but `config:check` seems to not actually check the values.  In my app-config.local.yaml (notice misspelling of gitlab.visibility value):
```yaml
scaffolder:
  github:
    visibility: private # 'public' or 'internal' or 'private'                                                      
  gitlab:
    visibility: publix # 'public' or 'internal' or 'private'                                                       
  # bitbucket:                                                                                                     
  #   visibility: public # 'public' or 'private'                                                                   
```

```bash
backstage$ yarn backstage-cli config:check
yarn run v1.22.1
$ /Users/alan/src/backstage/node_modules/.bin/backstage-cli config:check
Loaded config from app-config.yaml, app-config.local.yaml
✨  Done in 2.06s.
backstage$ 
```

**Review by freben — APPROVED** [review]

**Pull request merged**